code/game/objects/structures/salvaging.dm

```diff
@@ -0,0 +1,659 @@
+		/obj/item/stack/sheet/mineral/uranium/five = 5,
+		/obj/item/stack/sheet/mineral/diamond/five = 1
+	)
+
+/obj/structure/salvageable/circuit_imprinter
+	name = "broken circuit imprinter"
+	icon_state = "wreck_circuit_imprinter"
+	salvageable_parts = list(
+		/obj/item/stack/sheet/glass/two = 80,
+		/obj/item/stack/cable_coil/cut = 80,
+		/obj/item/stack/ore/salvage/scrapuranium/five = 60,
+		/obj/item/stack/ore/salvage/scrapmetal/five = 60,
+		/obj/item/stack/ore/salvage/scrapbluespace = 60,
+
+		/obj/effect/spawner/lootdrop/salvage_matter_bin = 40,
+		/obj/effect/spawner/lootdrop/salvage_manipulator = 30,
+
+		/obj/item/stack/circuit_stack = 50, //this might be the only way in the game to get a poly circuit, and the only way for many ships to get essensial electronics. huh.
+		/obj/effect/spawner/lootdrop/random_machine_circuit_mech = 45, //with all the wonderful broken mechs lying around, this might be a chance to get something stupidly overpowered.
+		/obj/effect/spawner/lootdrop/random_machine_circuit_common = 50, //well.... "common"
+		/obj/effect/spawner/lootdrop/random_machine_circuit_rare = 5,
+
+		/obj/item/stack/sheet/metal/five = 15, //same as above but more geared towards stuff used by circuit imprinter
+		/obj/item/stack/sheet/glass/five = 15,
+		/obj/item/stack/sheet/mineral/silver/five = 15,
+		/obj/item/stack/sheet/mineral/gold/five = 15,
+		/obj/item/stack/sheet/bluespace_crystal/five = 5,
+		/obj/item/stack/sheet/mineral/diamond/five = 1
+	)
+
+/obj/structure/salvageable/destructive_analyzer
+	name = "broken destructive analyzer"
+	desc = "If this thing could power up, it would probably slice you in half. You may be able to salvage something from this." //this ones pretty dangerous
+	icon_state = "wreck_d_analyzer"
+	salvageable_parts = list(
+		/obj/item/stack/sheet/glass/two = 80,
+		/obj/item/stack/cable_coil/cut = 80,
+		/obj/item/stack/ore/salvage/scrapuranium/five = 60,
+		/obj/item/stack/ore/salvage/scrapmetal/five = 60,
+		/obj/item/stack/ore/salvage/scrapplasma = 60,
+
+		/obj/effect/spawner/lootdrop/salvage_scanning = 40,
+		/obj/effect/spawner/lootdrop/salvage_laser = 30,
+		/obj/effect/spawner/lootdrop/salvage_manipulator = 30,
+
+		/obj/item/storage/toolbox/syndicate/empty = 80,
+		/obj/effect/spawner/lootdrop/destructive_anal_loot = 65,
+
+		/obj/item/stack/sheet/metal/five = 15, //same as above but more geared towards stuff used by circuit imprinter
+		/obj/item/stack/sheet/glass/five = 15,
+		/obj/item/stack/sheet/mineral/silver/five = 15,
+		/obj/item/stack/sheet/mineral/gold/five = 15,
+		/obj/item/stack/sheet/bluespace_crystal/five = 5,
+		/obj/item/stack/sheet/mineral/diamond/five = 1
+	)
+
+/obj/structure/salvageable/destructive_analyzer/dismantle(mob/living/user)
+	. = ..()
+	var/danger_level = rand(1,100)
+	switch(danger_level) //scary.
+		if(60 to 100)
+			src.audible_message("<span class='notice'>You can hear the sound of broken glass in the [src].</span>")
```

Comment:
src.proc is redundant, the [src] is also redundant on all of these

Reply:
removed the src.